### PR TITLE
Bug 1216564 - Fix launch{_packaged}_app.py tests in order to support the new homescreen

### PR DESCRIPTION
**Pull request by mwargers**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=356f6371941e8f9751e97632b99f53c77afb1e99

File: tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py [hunk 1]
```diff
@@ -177,24 +174,12 @@ def number_of_columns(self):
 
     class InstalledApp(PageRegion):
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py]: We don't need switch_to_root_element() in this class, which makes it way simpler. Cool! :)

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a9a00c77e8516aee3ceac81dccec62ac96e25fab

File: tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py [hunk 2]
```diff
@@ -26,36 +26,27 @@ class Homescreen(Base):
...
-        return SearchPanel(self.marionette)
+    def launch(self):
+        Base.launch(self)
 
-    def wait_for_app_icon_present(self, app_name):
-        Wait(self.marionette, timeout=30).until(lambda m: self.installed_app(app_name))
+    def wait_for_app_icon_present(self, app_manifest):
+        Wait(self.marionette, timeout=30).until(lambda m: self.installed_app(app_manifest))
 
     def wait_for_app_icon_not_present(self, app_name):
```
Comment (JohanLorenzo) [inline tests/python/gaia-ui-tests/gaiatest/apps/homescreen/app.py]: Nit: Change to `app_manifest`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2f502953292f64ac0a9ad81258c1f915e9e5826c

**Pull request merged**